graphql/database/userImpact/UserImpactModel.js

```diff
@@ -0,0 +1,88 @@
+import BaseModel from '../base/BaseModel'
+import types from '../fieldTypes'
+import tableNames from '../tables'
+import { USER_IMPACT, USER_VISIT_IMPACT_VALUE } from '../constants'
+import { permissionAuthorizers } from '../../utils/authorization-helpers'
+
+/*
+ * @extends BaseModel
+ */
+class UserImpact extends BaseModel {
+  static get name() {
+    return USER_IMPACT
+  }
+
+  static get hashKey() {
+    return 'userId'
+  }
+
+  static get rangeKey() {
+    return 'charityId'
+  }
+
+  static get tableName() {
+    return tableNames.userImpact
+  }
+
+  static get schema() {
+    const self = this
+    return {
+      userId: types
+        .string()
+        .required()
+        .description(
+          `The unique user ID from our authentication service (Firebase)`
+        ),
+      charityId: types
+        .string()
+        .required()
+        .description(`the charity id that maps to the charity table`),
+      userImpactMetric: types
+        .number()
+        .default(self.fieldDefaults.userImpactMetric)
+        .description(
+          `the metric a user contributes to a charity.  For example, it might be number of treats for
+          cats.`
+        ),
+      pendingUserReferralImpact: types
```

Comment:
Do we also need `pendingUserReferrals` to know the number of referrals to display?

Reply:
yeah I was assuming we'd deduce that based on impact but it makes sense to store both